gym_solo/core/test_termination_factory.py

```diff
@@ -0,0 +1,62 @@
+import unittest
+from gym_solo.core import termination
+
+class DummyTermination(termination.Termination):
+  def __init__(self, body_id: int, termination_var: bool):
+    self.body_id = body_id
+    self.termination_var = termination_var
+    self.reset_counter = 0
+    self.reset()
+    
+  def reset(self):
+    self.reset_counter += 1
+
+  def is_terminated(self) -> bool:
+    return self.termination_var
+
+class TestTerminationFactory(unittest.TestCase):
+  def test_initialization(self):
+    termination_factory = termination.TerminationFactory()
+    self.assertFalse(termination_factory._terminations)
+    self.assertTrue(termination_factory._use_or)
+
+  def test_register_termination(self):
+    termination_factory = termination.TerminationFactory()
+    dummy_termination_1 = DummyTermination(0, True)
+    dummy_termination_2 = DummyTermination(0, False)
+    termination_factory.register_termination(dummy_termination_1, 
+      dummy_termination_2)
+    self.assertTrue(len(termination_factory._terminations) == 2)
```

Comment:
Use `assertEqual` instead between the length and `2`? the main reason is if the test fails, in the error message it'll be easier to understand what went wrong